### PR TITLE
Add job and log clients

### PR DESCRIPTION
**Pull request by acroz**

### CODE REVIEW

File: sherlockml/clients/job.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+class JobClient(BaseClient):
+
+    SERVICE_NAME = "steve"
+
+    def list(self, project_id):
+        endpoint = "/project/{}/job".format(project_id)
+        return self._get(endpoint, JobSummarySchema(many=True))
+
+    def create_run(self, project_id, job_id, parameter_value_sets):
```
Comment (pbugnion) [inline sherlockml/clients/job.py]: Should there be a docstring for creating a run? In particular, the structure of `parameter_value_sets` is not obvious.

Comment (pbugnion) [reply sherlockml/clients/job.py]: Not sure if it's conventional in this, but should we consider a default `parameter_value_sets=None` (obviously defaulting to `[{}]`)?

**Comment by pbugnion** [comment]
Should the client for job logs be called just `LogClient`? Feels like it precludes adding log clients for other things.

**Comment by pbugnion** [comment]
Looks good, modulo some comments inline.

**Comment by acroz** [comment]
> Should the client for job logs be called just `LogClient`? Feels like it precludes adding log clients for other things.

I discussed this with others, and we came to the agreement that it was likely that other centralised log storage in the future would more likely be with Wozniak than not, and that calling this simply `LogClient` was suitable.

**Pull request merged**